src/plugin.validation.dns.godaddy/Godaddy/DnsManagementClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using PKISharp.WACS.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PKISharp.WACS.Plugins.ValidationPlugins.Godaddy
+{
+
+    public class DnsManagementClient
+    {
+        private readonly string _apiKey;
+        private readonly ILogService _logService;
+        private readonly string uri = "https://api.godaddy.com/";
+
+        public DnsManagementClient(string apiKey, ILogService logService)
+        {
+            _apiKey = apiKey;
+            _logService = logService;
+        }
+
+        public async Task CreateRecord(string record, string identifier, RecordType type, string value)
+        {
+            using (var client = new HttpClient { BaseAddress = new Uri(uri) })
+            {
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                client.DefaultRequestHeaders.Add("Authorization", $"sso-key {_apiKey}");
+
+                var putData = new List<object>() { new { name = "_acme-challenge", ttl = 3600, data = value } };
+
+                string serializedObject = Newtonsoft.Json.JsonConvert.SerializeObject(putData);
+
+                //Record successfully created
+                // Wrap our JSON inside a StringContent which then can be used by the HttpClient class
+                var typeTxt = type.ToString();
+                var httpContent = new StringContent(serializedObject, Encoding.UTF8, "application/json");
+                var buildApiUrl = $"v1/domains/{identifier}/records/{typeTxt}";
```

Comment:
Does GoDaddy always have all DNS records of a registerable domain in one zone, or is it possible to create multiple zones or host subdomains there as well?